builder/parser/line_parsers.go

```diff
@@ -195,6 +192,44 @@ func parseLabel(rest string) (*Node, map[string]bool, error) {
 	return parseNameVal(rest, "LABEL")
 }
 
+// parses a statement with multiple keywords definition and name-value
+// assignment, like `name1= name2=value`.
+// Note that this is a stricter format than the old format of assignment, in a way
+// that it only allows assignment of the form `name=value`. And that a name
+// definition or assignment to an empty value is designated by `name=`.
+func parseNameOrNameVal(rest string) (*Node, map[string]bool, error) {
+	words := parseWords(rest)
+	if len(words) == 0 {
+		return nil, nil, nil
+	}
+
+	var (
+		rootnode *Node
+		prevNode *Node
+	)
+	for i, word := range words {
+		if !strings.Contains(word, "=") {
+			return nil, nil, fmt.Errorf("Syntax error - can't find = in %q. Must be of the form: name=value", word)
```

Comment:
While playing with this I think printing the error here might not be ideal.
I see this output:
```
docker build tmp
Sending build context to Docker daemon 40.07 MB
Error response from daemon: Syntax error - can't find = in "foo". Must be of the form: name=value
```
For `ARG foo` in my Dockerfile.
The problem is that there's nothing in the error message to tell me that the error is related to a bad `ARG` command.
Would it make more sense to have the parser accept "foo", "foo=" and "foo=bar" but then have the arg() func, in dispatcher.go, check for a "=" and generate a more focused error message?  Additionally, I'd prefer if the parser wasn't this smart about what it was parsing. We should be moving away from a smart parser (that knows about Dockerfile commands) and one that just knows about words/spaces/etc..